Support setting Tuple, Dict, and Union to empty in the command line.

ShenhanQian opened the pull request:
Hi Brent, thanks for your kind suggestions on issue #56!

This commit makes it possible to set Tuple, Dict, and Union to empty by passing in `-` in the command line as the only argument. The functionality is tested with an additional test in `test_dcargs.py`.

Note that I'm currently using `-` instead of `*` as the indicator of an empty container because `*` is usually expanded into the file names in the working directory, in which case the user has to pass in `"*"` to disable the expansion.

For the example below,
```python
# main.py
from dataclasses import dataclass

@dataclass()
class BaseConfig:
    x: tuple[str, ...] = ("a", "b", "c")

if __name__ == "__main__":
    print(tyro.cli(BaseConfig).x)
```
we can now set the item `x` to empty with the following command:
```shell
python main.py --x -
```
Output:
```shell
()
```

brentyi commented:
Thanks for the PR! The looks slightly different from what I hand in mind but overall reasonable. 🙂 

As a question on usage,  do you have a preference between:
```
python main.py --x -
```
vs just
```
python main.py --x
```

for setting `x` to an empty list?

When I had mentioned `"*"` in the original issue I meant in a slightly more literal since, where we're currently passing in `nargs="+"` we could instead pass in `nargs="*"`. (context: https://docs.python.org/3/library/argparse.html#nargs)

ShenhanQian commented:
Thanks for clarifying. I also prefer `python main.py --x` for its simplicity.

In the lastest commit, I have replaced all "+" for "*" and updated the test accordingly. 

I also notice that we cannot clear a non-empty dict in the command line because the command-line interface already decouples the dict into sub items.

```python
def test_empty_container() -> None:
    @dataclasses.dataclass
    class A:
        x: Tuple[int, ...] = (1, 2, 3)
        y: Union[int, str, List[bool]] = dataclasses.field(default_factory=lambda: [1, 2, 3])
        y: Dict[str, int] = dataclasses.field(default_factory=lambda: {'a': 1})

    assert tyro.cli(A, args="--x".split(" ")).x == ()  # succeed
    assert tyro.cli(A, args="--y".split(" ")).y == []  # succeed
    assert tyro.cli(A, args="--z".split(" ")).z == {}  # fail
```

```
usage: test_dcargs.py [-h] [--x INT [INT ...]] [--y INT|STR|{{True,False} [{True,False} ...]}] [--z.a INT]


test_dcargs.py: error: unrecognized arguments: --z
```
